GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1687,18 +1691,11 @@ private void RevisionsCellPainting(object sender, DataGridViewCellPaintingEventA
                         int gravatarTop = e.CellBounds.Top + textHeight + 6;
                         int gravatarLeft = e.CellBounds.Left + baseOffset + 2;
 
-
-                        Image gravatar = Gravatar.GravatarService.GetImageFromCache(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, FallBackService.MonsterId);
-
-                        if (gravatar == null && !string.IsNullOrEmpty(revision.AuthorEmail))
-                        {
-                            ThreadPool.QueueUserWorkItem(o =>
-                            Gravatar.GravatarService.LoadCachedImage(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, null, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, RefreshGravatar, FallBackService.MonsterId));
-                        }
-
+                        var gravatar = await _gravatarService.GetAvatarAsync(revision.AuthorEmail, gravatarSize, DefaultImageType.MonsterId);
                         if (gravatar != null)
+                        {
                             e.Graphics.DrawImage(gravatar, gravatarLeft + 1, gravatarTop + 1, gravatarSize, gravatarSize);
```

Comment:
I've rejigged few things around:
- the rendering is now synchronous
- the image is loaded synchronously from the local cache
- if the image isn't in the local cache - an unblocking call is scheduled for the service to download the image

I think I also know why the cardview never worked for me - the gravatar service would be blocking the rendering trying to download images over disconnected network.